mlflow/store/sqlalchemy_store.py

```diff
@@ -463,6 +463,26 @@ def set_tag(self, run_id, tag):
             self._check_run_is_active(run)
             session.merge(SqlTag(run_uuid=run_id, key=tag.key, value=tag.value))
 
+    def delete_tag(self, run_id, key):
+        """
+        Delete a tag from a run.
+        :param run_id: ID of the run that contains the tag to delete
+        :param key: name of the tag to delete
+        """
+        with self.ManagedSessionMaker() as session:
+            run = self._get_run(run_uuid=run_id, session=session)
+            self._check_run_is_active(run)
+            filtered_tags = session.query(SqlTag).filter_by(run_uuid=run_id, key=key).all()
+            if len(filtered_tags) == 0:
+                raise MlflowException(
+                    "No tag with name: {} in run with id {}".format(key, run_id),
+                    error_code=RESOURCE_DOES_NOT_EXIST)
+            elif len(filtered_tags) > 1:
+                raise MlflowException(
+                    "Bad data - tag with multiple value entries. Please file an issue.",
```

Comment:
It should be sufficient to say bad data in database. This can happen only due to some corruption or duplicate tag is entered in database table. Instead of asking users to file issue, can we redirect to documentation. If the docs don't say that the tags should be unique let's fix that.

Reply:
There are currently no docs that I can find that express this clearly. I've added a sentence to the section about adding tags to MLFlow in the docs and have updated the exception message to direct to that, though I think this should really only happen if there's a bug in MLFlow and if there is, an issue should be filed.